modules/utils/src/merkle.ts

```diff
@@ -1,26 +1,36 @@
+import * as merkle from "vector-merkle-tree";
 import { CoreTransferState } from "@connext/vector-types";
 import { HashZero } from "@ethersproject/constants";
 import { keccak256 } from "ethereumjs-util";
 import { MerkleTree } from "merkletreejs";
 
-import { hashCoreTransferState } from "./transfers";
-
-export const generateMerkleTreeData = (transfers: CoreTransferState[]): { root: string; tree: MerkleTree } => {
-  // Sort transfers alphabetically by id
-  const sorted = transfers.sort((a, b) => a.transferId.localeCompare(b.transferId));
+import { encodeCoreTransferState, hashCoreTransferState } from "./transfers";
 
+export const generateMerkleTreeData = (transfers: CoreTransferState[]): { root: string; tree: merkle.Tree } => {
   // Create leaves
-  const leaves = sorted.map((transfer) => {
-    return hashCoreTransferState(transfer);
+  const tree = new merkle.Tree();
+  tree.free(); // handle memory leaks
```

Comment:
You need to call `free` after you're done using the Tree, not right after construction.

Example:
```typescript
const tree = new merkle.Tree();
let calculated;
try {
   // TODO: Insert hex here
   calculated = tree.root_js();
} finally {
   tree.free();
}